lib/sycamore/sycamore/query/compare.py

```diff
@@ -0,0 +1,66 @@
+from typing import Any, Set
+
+from sycamore.query.logical_plan import LogicalPlan, Node
+
+
+def compare_graphs(node_a: Node, node_b: Node, visited_a: Set[int], visited_b: Set[int]) -> dict[str, Any]:
```

Comment:
Could we just use https://dictdiffer.readthedocs.io/en/latest/ ?

Reply:
reading the link the PR seems to give a little more control on what we're comparing. I also prefer being able to access the class.compare() function